pkgs/native_assets_builder/lib/src/package_layout/package_layout.dart

```diff
@@ -29,26 +30,30 @@ class PackageLayout {
 
   final Uri packageConfigUri;
 
-  PackageLayout._(
-      this.rootPackageRoot, this.packageConfig, this.packageConfigUri);
+  PackageLayout._(this._fileSystem, this.rootPackageRoot, this.packageConfig,
+      this.packageConfigUri);
 
   factory PackageLayout.fromPackageConfig(
+    FileSystem fileSystem,
     PackageConfig packageConfig,
     Uri packageConfigUri,
```

Comment:
Ditto: We could consider using `File packageConfig` which would encapsulate both the `packageConfigUri` and the `fileSystem`.

Reply:
See my earlier comment. I think it's cleaner if public APIs receive a `FileSystem` and `Uri`s instead of receiving `File` / `Directory` objects that we
a) extract the higher-level `FileSystem` out of
b) create assumptions in the code base that all `Directory` and `File`s we get are from same `FileSystem`